hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -216,9 +216,10 @@ private int getLogBlockLength(int contentLength, int headerLength, int footerLen
 
   private void rolloverIfNeeded() throws IOException {
     // Roll over if the size is past the threshold
-    if (getCurrentSize() > sizeThreshold) {
-      LOG.info("CurrentSize " + getCurrentSize() + " has reached threshold " + sizeThreshold
-          + ". Rolling over to the next version");
+    long currentSize = fs.getFileStatus(logFile.getPath()).getLen();
+    if (currentSize > sizeThreshold) {
```

Comment:
@prashantwason @suryaprasanna : Can you guys review this patch. surprised that rolliover isn't working.

Reply:
Sorry, I found it has been fixed in version 0.12.1 or above by [HUDI-4780](https://issues.apache.org/jira/browse/HUDI-4780).